subsystem/kubeapi_test.go

```diff
@@ -119,7 +141,7 @@ func waitForClusterDeploymentCRDState(
 	start := time.Now()
 	for time.Duration(timeout)*time.Second > time.Since(start) {
 		clusterDeployment = getClusterDeploymentCRD(ctx, client, key)
-		if clusterDeployment.Status.Conditions[0].Message == state {
+		if len(clusterDeployment.Status.Conditions) > 0 && clusterDeployment.Status.Conditions[0].Message == state {
```

Comment:
i'm not sure that the order here is guarantied, probably need to search for specific condition.

Reply:
There are helper functions, for example:
`conditionsv1.FindStatusCondition`

Reply:
Even better :) thanks @rollandf

Reply:
Hive clusterDeployment conditions are not implemented with with @djzager's [openshift/custom-resource-status](https://github.com/openshift/custom-resource-status), so I'll copy over FindStatusCondition as an interim solution.

For installEnv I did use [openshift/custom-resource-status](https://github.com/openshift/custom-resource-status) so I can use the search func directly from there.
Thanks!

Reply:
Done.